CONTRIBUTING.md

```diff
@@ -0,0 +1,152 @@
+
+## Maintainership
+
+All repositories should have a [CODEOWNERS](codeowners) file included with a
+specified owner for the project. Preferably a team or 2+ individuals would be
+marked as "owners" who are responsible for the project. These owners are
+responsible for the maintainership of the project and controlling the scope of
+features within the said project. They shall have the final say on whether or not a
+feature request is a good fit for the scope of a project.
+
+## Best Practices
+
+The following are best practices that we strive for in all of our projects. A
+project may override these where it makes sense. The respective project should
+have an updated `CONTRIBUTING.md` included with information about best practices
+for the project.
+
+### Branch Names
+
+The default branch for repositories is `trunk`. Always work from the `trunk`
+branch and open up pull requests against it.
+
+Projects should always reference a tagged version of a project and never track
+`trunk`/`dev-trunk`. The `trunk` branch is considered 'nightly' and unstable at
+all times and is subject to change without notice. Releases that follow the
+[release process](#release-process) should always be backward compatible and
+well documented via a `CHANGELOG`.
+
+### Coding Style
+
+If the project maintainer has any additional requirements, you will find them
+listed here.
+
+#### WordPress Coding Standard
+
+Code must generally follow the [WordPress Coding
+Standard](wordpress-coding-standard). Our projects follow the standards via a
+[PHPCS ruleset](phpcs-ruleset) with some modifications from the core standards
+to allow for some added flexibility.
+
+#### Tests
+
+Contributions to projects should have tests associated with them. Failure to
+include tests could result in your contribution being declined.
+
+#### Backward Compatibility
+
+All contributions must be backward compatible unless a breaking change is
+being made intentionally. Breaking changes must follow [semantic
+versioning](semvar) and publish a new major release with a call out to the
+breaking change in the project's CHANGELOG.
+
+#### Documentation
+
+Ensure that the `README.md`, `CHANGELOG`, and any other documentation are
+up-to-date with any changes made.
+
+#### Compiled Assets
+
+Repositories should not contain compiled/built assets within their main
+branches. This includes minified CSS or built Javascript files. Repositories
+should use a GitHub action to produce a `*-built` branches and tags. These built
+branches/tags can be used as submodules or downloaded directly for external use.
+
+You can use a [Built Branch](built-branch) or [Built Tag](built-tag) action to
+automatically compile your Composer/Node dependencies into `*-built` branches
+and tags.
+
+### Release Process
+
+The following is a release process for all projects. All projects must follow
+[semantic versioning](semvar).
+
+1. Starting from the default branch of the repository, create a release branch
```

Comment:
I wonder if it will be confusing to people who would expect their initial releases to be a pre-stable, e.g. `0.1.0`, release number?

Reply:
They should still have to create some form of a release branch be it `release/0.1.0`. I follow that process on Mantle to cut a new release and update the files (CHANGELOG, plugin, etc.)

Reply:
Agreed, and it's probably fine. I was just thinking out loud about whether someone will take that version number too literally and start right at `release/1.0.0`.

Reply:
Maybe it should be `X.X.X`?

Reply:
I like that suggestion.